tests/virt_autotest/login_console.pm

```diff
@@ -40,14 +40,23 @@ sub login_to_console {
     $timeout //= 5;
     $counter //= 240;
 
+    reset_consoles;
+    select_console 'sol', await_console => 0;
+
     if (check_var('PERF_KERNEL', '1') or check_var('CPU_BUGS', '1') or check_var('VT_PERF', '1')) {
-        reset_consoles;
-        select_console 'sol', await_console => 0;
-        send_key_until_needlematch(['linux-login', 'virttest-displaymanager'], 'ret', $counter, $timeout);
-        #use console based on ssh to avoid unstable ipmi
-        save_screenshot;
-        use_ssh_serial_console;
-        return;
+        if (get_var("XEN")) {
+            assert_screen 'pxe-qa-net-mitigation', 90;
+            send_key 'ret';
+            assert_screen([qw(grub2 grub1)], 60);
```

Comment:
if you want to catch up grub screen which is before login prompt . should put this code above login needles assert code.